src/table/body-cell/index.tsx

```diff
@@ -2,72 +2,127 @@
 // SPDX-License-Identifier: Apache-2.0
 import clsx from 'clsx';
 import styles from './styles.css.js';
-import React from 'react';
+import React, { useCallback, useRef } from 'react';
+import useFocusVisible from '../../internal/hooks/focus-visible';
+import { useEffectOnUpdate } from '../../internal/hooks/use-effect-on-update';
+import Button from '../../button/internal';
+import { ButtonProps } from '../../button/interfaces';
 import { TableProps } from '../interfaces';
-import { useVisualRefresh } from '../../internal/hooks/use-visual-mode';
+import { TableTdElement, TableTdElementProps } from './td-element';
+import { InlineEditor } from './inline-editor';
+import { useStableScrollPosition } from './use-stable-scroll-position';
 
-interface TableBodyCellProps {
-  className?: string;
-  style?: React.CSSProperties;
-  wrapLines: boolean | undefined;
-  isFirstRow: boolean;
-  isLastRow: boolean;
-  isEvenRow?: boolean;
-  stripedRows?: boolean;
-  isSelected: boolean;
-  isNextSelected: boolean;
-  isPrevSelected: boolean;
-  children?: React.ReactNode;
-  hasSelection?: boolean;
-  hasFooter?: boolean;
+const readonlyState = Object.freeze({
+  isEditing: false,
+  currentValue: undefined,
+  setValue: () => {},
```

Comment:
This one can be `istanbul-ignore-next` because it only exists to satisfy types and should never be called for real